CMS-Enterprise/easi-app
# [ EASI-4647] Okta session check - Auto login and redirect

# EASI-4647

## Description


- Added hook in `useOktaSession` to check if okta session exists via Okta api
- Implemented session check on `<WelcomePage />`.  
  - If session exists, replace router link with button that calls `signInWithRedirect`
- Implemented session check on `<OktaSignInWidget />`.  
  - While checking session disable widget with an overlay and loading spinner
  - If session exists immediately `signInWithRedirect` to authenticate within app
- Unauthenticated redirects should still work as expected within the app
  - For example, hitting `/trb` while unauthenticated will redirect to `/signin` then check for existing okta session.  If exists, it will redirect you back to `/trb`
- Addressed existing flicker within initial app load of `<UserInfoWrapper />` by checking of `authState` has loaded.  
  - If there is no local `authState`, and an Okta session does not exist, render the `<WelcomePage />` page.
- If user is logged out of their session in another environment - `https://test.easi.cms.gov/`, the user will not be logged out of EASI locally, as the `authState` still persists on the user's machine and takes priority over the Okta session
  - The same holds true if a user logs out of EASI locally, the test environment will remain logged in until logged out manually or app times out
- Sessions are shared, however each device has their own `authState`, which takes priority in the local app.
- After logging out, session will not be restored upon refresh as the session does not exist.  The user will need to use the widget login mechanism to start a new session
- Added a `display: none` class to `<PageWrapper />` component as to not flicker when redirecting.  We still want to process the component, just not render


## How to test this change

1.  Log out of EASI locally and in test env - `https://test.easi.cms.gov/`
2. Log into test env
3. Load the app/refresh EASI
4. Confirm that the app loads with the landing page.
5. Inspect the html to verify both sign in links on the landing page and header have been replace with buttons
6. Verify that clicking either will automatically authenticate the user
7. Log out of test
8. Verify user can still access EASI locally until manually signed out
9. Repeat the process but this time log out of EASI locally instead of test.
10. Verify that refreshes to no automatically sign the user in (as the session has been deleted)
11. Verify that test remains authenticated after refresh.
12. Repeat from step 1, except when attempting to hit the landing/home page, attempt to the the Help and Knoledge center directly
13. Verify redirect works after session check and authentication

## PR Author Checklist




## PR Reviewer Guidelines

- It's best to pull the branch locally and test it, rather than just looking at the code online!
- When approving a PR, provide a reason _why_ you're approving it
  - e.g. "Approving because I tested it locally and all functionality works as expected"
  - e.g. "Approving because the change is simple and matches the Figma design"
- Don't be afraid to leave comments or ask questions, especially if you don't understand why something was done! (This is often a great time to suggest code comments or documentation updates)
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
